src/gucs/planning.rs

```diff
@@ -11,6 +11,8 @@ pub static ENABLE_INDEX: GucSetting<bool> = GucSetting::<bool>::new(true);
 
 pub static SEARCH_MODE: GucSetting<Mode> = GucSetting::<Mode>::new(Mode::basic);
 
+pub static ENABLE_PGVECTOR_COMPATIBILITY: GucSetting<bool> = GucSetting::<bool>::new(false);
```

Comment:
Will there be any side effects if the default here is set to "on"?

Reply:
The only unknown issue is: for `pg12`, crash inside this hook could not recover due to pgrx design: https://github.com/tensorchord/pgvecto.rs/pull/256#discussion_r1449995690

However, as there is so many pointer operation, I am not very confident about whether there is UB not checked out. Marking it as default behavior might be a bit aggressive.

Reply:
Understand. We can keep it off as default